sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4138,7 +4138,7 @@ class SQLConf extends Serializable with Logging {
 
   def enforceReservedKeywords: Boolean = ansiEnabled && getConf(ENFORCE_RESERVED_KEYWORDS)
 
-  def ansiFailOnElementNotExists: Boolean = ansiEnabled && getConf(ANSI_STRICT_INDEX_OPERATOR)
+  def failOnNonexistentElement: Boolean = ansiEnabled && getConf(ANSI_STRICT_INDEX_OPERATOR)
```

Comment:
shall we follow the name of the config? `strictIndexOperator`

Reply:
I think `failOnNonexistentElement` is fine - it signifies the circumstance (in case of non-existent element) where failure would happen.

Reply:
Let's use `strictIndexOperator` instead. The new configuration doesn't control the function `element_at`/`elt` when the index is out of array bound.